### PR TITLE
Change versioning scheme

### LINKED ISSUE #1138: Deemphasize upstream version number
daira
> Proposal: decouple Zcash versions from upstream versions, and call the next release Zcash v0.8 rather than v0.11.2.z8
> The launch release would be Zcash v1.0 (with the usual beta/release candidate version structure)
> Hmm, but 0.8 < 0.11.2. Any suggestions to fix that?
> I guess we could keep the v0.11.2.z* pattern until 1.0 beta 1 (which would be the first release that supports the payment API).
> Everyone refers to the releases as just z* anyway, but I do feel that the full version number gives undue weight to the upstream version we forked from.

jackgavigan
> v0.12

jack
> -1 on v0.12, that's going to get too confusing

daira
> On reflection I think "keep the v0.11.2.z* pattern until the payment API release" is a better approach. And then going straight to 1.0 beta 1 will avoid any confusion with Bitcoin releases. I don't know what we'll do with the version number APIs; we'll have to think about that in advance so that changing them doesn't block that release.

### PR DESCRIPTION
**Pull request by str4d**
Closes #1138

### CODE REVIEW

File: doc/release-process.md [hunk 1]
```diff
@@ -21,8 +22,18 @@ Also, the following commands use the ZCASH_RELEASE_PREV bash variable for the pr
...
     configure.ac
     
-In `configure.ac` and `clientversion.h` change CLIENT_VERSION_IS_RELEASE to
-false while Zcash is in alpha-test phase.
+In `configure.ac` and `clientversion.h`:
+
+- Increment `CLIENT_VERSION_BUILD` according to the following schema:
+
+  - 0-24: `1.0.0-beta1`-`1.0.0-beta25`
+  - 25-49: `1.0.0-rc1-1.0.0-rc25`
```
Comment (daira) [inline doc/release-process.md]: `1.0.0-rc1`-`1.0.0-rc25`

**Comment by daira** [comment]
ACK modulo https://github.com/zcash/zcash/pull/1468/files#r81461219

**Comment by str4d** [comment]
@zkbot r+

**Comment by daira** [comment]
```
test/util_tests.cpp(354): error in "test_FormatSubVersion": check FormatSubVersion("Test", 99900, std::vector<std::string>()) == std::string("/Test:0.9.99/") failed [/Test:0.9.99-beta1/ != /Test:0.9.99/]
test/util_tests.cpp(355): error in "test_FormatSubVersion": check FormatSubVersion("Test", 99900, comments) == std::string("/Test:0.9.99(comment1)/") failed [/Test:0.9.99-beta1(comment1)/ != /Test:0.9.99(comment1)/]
test/util_tests.cpp(356): error in "test_FormatSubVersion": check FormatSubVersion("Test", 99900, comments2) == std::string("/Test:0.9.99(comment1; comment2)/") failed [/Test:0.9.99-beta1(comment1; comment2)/ != /Test:0.9.99(comment1; comment2)/]
```

**Comment by daira** [comment]
I have a fix.

**Comment by daira** [comment]
@str4d: https://github.com/str4d/zcash/pull/4

**Comment by str4d** [comment]
@zkbot r+

**Pull request merged**